File: core/core-domain/src/main/java/com/coffee/gu/coupon/IssuedCouponRepository.java
package com.coffee.gu.coupon;

import java.util.List;
import java.util.Optional;

public interface IssuedCouponRepository {
    IssuedCoupon save(IssuedCoupon issuedCoupon);
    List<IssuedCoupon> saveAll(List<IssuedCoupon> issuedCoupons);
    List<IssuedCoupon> findAllByPrincipalKey(String principalKey);
    boolean existsByPrincipalKeyAndCouponId(String principalKey, Long couponId);
    Optional<IssuedCoupon> findById(Long issuedCouponId);
    List<IssuedCoupon> findAllByIdIn(List<Long> issuedCouponIds);

}

File: core/core-domain/src/main/kotlin/com/coffee/gu/coupon/IssuedCouponRepository.kt
package com.coffee.gu.coupon

interface IssuedCouponRepository {
    fun save(issuedCoupon: IssuedCoupon): IssuedCoupon
    fun saveAll(issuedCoupons: List<IssuedCoupon>): List<IssuedCoupon>
    fun findAllByPrincipalKey(principalKey: String): List<IssuedCoupon>
    fun existsByPrincipalKeyAndCouponId(principalKey: String, couponId: Long): Boolean
    fun findById(issuedCouponId: Long): IssuedCoupon?
    fun findAllByIdIn(issuedCouponIds: List<Long>): List<IssuedCoupon>
}
